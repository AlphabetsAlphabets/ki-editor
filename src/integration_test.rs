#[cfg(test)]
mod integration_test {
    use std::time::Duration;

    use crossterm::event::Event;

    use crate::{
        canonicalized_path::CanonicalizedPath, key_event_parser::parse_key_events, screen::Screen,
    };

    // #[test]
    fn lsp_completion() {
        let mut screen = Screen::new().unwrap();
        let (sender, receiver) = std::sync::mpsc::channel();
        std::thread::sleep(Duration::from_secs(5));
        for event in parse_key_events("c-q").unwrap() {
            sender.send(Event::Key(event)).unwrap();
        }
        screen.run(None, receiver).unwrap();
    }
}
