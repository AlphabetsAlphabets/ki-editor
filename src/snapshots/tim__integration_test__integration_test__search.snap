---
source: src/integration_test.rs
expression: test_runner.content()
---
<Untitled> [NORMAL:CUSTOM]                                                      
1│pub struct Foo {                                                              
2│    a: (),                                                                    
3│    b: (),                                                                    
4│}                                                                             
5│                                                                              
6│pub fn foo() -> Foo {                                                         
7│    Foo { a: (), b: () }                                                      
8│}                                                                             
9│                                                                              
 │                                                                              
 │                                                                              
 │                                                                              
 │                                                                              
 │                                                                              
 │                                                                              
 │                                                                              
 │                                                                              
 │                                                                              
 │                                                                              
 │                                                                              
 │                                                                              
 │                                                                              
 │                                                                              
